src/coord/src/persistence.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
```

Comment:
Module level docs would help! I've been ask to look at compaction and trace/batch implementations, and I'm reading each line of each file to try and find them.

Reply:
I've heard this feedback loud and clear and am adding module level docs! resolving this comment just to clean up some pr state.